### PR TITLE
Added query parameters to parseUrl()

### PR DESCRIPTION
**Pull request by forbesjo**
This enhancement is to add query parameters to the object returned by parseUrl(). I think this would be useful to avoid duplicate query parameter code.

### CODE REVIEW

File: src/js/lib.js [hunk 1]
```diff
@@ -686,7 +686,25 @@ vjs.getAbsoluteURL = function(url){
...
+    getQueryParameters = function(url) {
+      var parameters = {}, urlParameters, name, value;
+
+      urlParameters = url.split('?');
+      if (urlParameters.length !== 2) {
+        return undefined;
+      }
+
+      urlParameters = urlParameters[1].split('&');
+      urlParameters.map(function(urlParameter) {
```
Comment (gkatsev) [inline src/js/lib.js]: `map` isn't available in IE8.

File: src/js/lib.js [hunk 2]
```diff
@@ -727,6 +745,10 @@ vjs.parseUrl = function(url) {
     document.body.removeChild(div);
   }
 
+  if(getQueryParameters(url)){
```
Comment (gkatsev) [inline src/js/lib.js]: We should cache the return value from this, so, we don't process it twice.

**Comment by gkatsev** [comment]
Why not have the `parseUrl` and `getQueryParameters` always return an object? Whether it's empty or not. This way, end users don't necessarily have to do an existence check for whether the object from `parseUrl` has a `parameters` property and only need to worry about whether they have the query param they really care for.
For example, this:
```js
var lang = videojs.parseUrl('http://example.com/video.mp4?language=en').parameters.language;
if (lang) {
  // do stuff with lang
}
```
Compared to this:
```js
var url = videojs.parseUrl('http://example.com/video.mp4?language=en').;
if (url.parameters && url.parameters.language) {
  // do stuff with url.parameters.language
}
```

**Comment by dmlap** [comment]
Browsers already do a thorough job of parsing `window.location`. Why not just parse the query parameters and leave it up to the native functionality for the rest?

**Comment by gkatsev** [comment]
Browsers don't parse query parameters. They leave it as a giant string on `window.location.search`, you still need to find your specific parameters from it.

**Comment by heff** [comment]
Does this need to be against stable and go out as a patch release, or can we switch it over to 5.x?

**Comment by gkatsev** [comment]
We have a bug that it would be nice if this is available. For 5.0, we could also use the [`qs`](https://www.npmjs.com/package/qs) module or just pass [`true` to node's `url.parse` as the second argument](https://nodejs.org/api/url.html#url_url_parse_urlstr_parsequerystring_slashesdenotehost) so it will parse the query params for us.

**Comment by heff** [comment]
Yeah, the `url.parse` route was what I was thinking too, for 5.0.

If there's a related bug in 4.x it'd be good to have a little more description about it here.

**Comment by dmlap** [comment]
I feel like a full URL parser is heading towards overkill. I know we're trying to provide more out-of-the-box for 5.0 but `window.location` really does the job for most non-query-param use-cases IMO.

**Comment by forbesjo** [comment]
In the BC player there's an incorrect regex for the language parameter, I thought that adding this enhancement instead of fixing the regex for that particular parameter would be more useful. 

If `url.parse` is going to be used for 5.0 then we can ignore this PR, I'll just use the modified regex and use the output from urlParse() when we upgrade to 5.0.

**Comment by forbesjo** [comment]
Closing, better to not reinvent the wheel.

**Pull request closed without merging**